Gordon360/Services/RecIM/SeriesService.cs

```diff
@@ -144,14 +147,43 @@ public async Task<SeriesViewModel> PostSeriesAsync(SeriesUploadViewModel newSeri
             return series;
         }
 
-        public SeriesScheduleViewModel GetSeriesScheduleByID(int seriesID)
+        public SeriesScheduleExtendedViewModel GetSeriesScheduleByID(int seriesID)
         {
             int scheduleID = _context.Series.Find(seriesID)?.ScheduleID ?? 0;
-            return _context.SeriesSchedule.Find(scheduleID);
+            SeriesScheduleExtendedViewModel res =  _context.SeriesSchedule.Find(scheduleID);
+            var surfaceIDs = _context.SeriesSurface.Where(ss => ss.SeriesID == seriesID).Select(s => s.SurfaceID);
+            res.SurfaceIDs = surfaceIDs;
+            return res;
         }
 
         public async Task<SeriesScheduleViewModel> PutSeriesScheduleAsync(SeriesScheduleUploadViewModel seriesSchedule)
         {
+
+            if (seriesSchedule.SeriesID is int seriesID)
+            {
+                var series = _context.Series.Find(seriesID);
+                //update surfaces
+                if (seriesSchedule.AvailableSurfaceIDs.Count() == 0) throw new UnprocessibleEntity { ExceptionMessage = "Schedule cannot have no surfaces" };
```

Comment:
```suggestion
                if (seriesSchedule.AvailableSurfaceIDs.Count() == 0) throw new UnprocessibleEntity { ExceptionMessage = "The schedule must have a surface" };
```
It's a little bit awkward and double-negative-ish